src/Helmet.js

```diff
@@ -187,7 +187,13 @@ const Helmet = (Component) => class HelmetWrapper extends React.Component {
                 return warn(`Only elements types ${VALID_TAG_NAMES.join(", ")} are allowed. Helmet does not support rendering <${child.type}> elements. Refer to our API for more information.`);
             }
 
-            if (nestedChildren && typeof nestedChildren !== "string") {
```

Comment:
Suggestion. Externalize the test into a function. That would make the code more declarative and easier to read on first glance and communicate better intent of the code.